jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -22,17 +22,17 @@ public static void premain(String agentArgument, Instrumentation instrumentation
        System.exit(1);
      }
 
+     String host;
      int port;
-     InetSocketAddress socket;
      String file;
 
      if (args.length == 3) {
        port = Integer.parseInt(args[1]);
-       socket = new InetSocketAddress(args[0], port);
+       host = args[0];
        file = args[2];
      } else {
        port = Integer.parseInt(args[0]);
-       socket = new InetSocketAddress(port);
+       host = "0.0.0.0";
```

Comment:
An empty string binds to the loopback interface.

I don't have an IPv6 only host, but on my machine it correctly binds on ipv4 and ipv6 interfaces.
as stated here https://github.com/eclipse/jetty.project/blob/0c8273f2ca1f9bf2064cd9c4c939d2546443f759/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNetworkConnector.java#L53

I also tried the special string "::" which is the quivalent of 0.0.0.0 for ipv6 and it does the same behavior